# HG changeset patch
# User Rik <rik@octave.org>
# Date 1425352716 28800
#      Mon Mar 02 19:18:36 2015 -0800
# Node ID fa290c42197eea45b9c6de2cdcb1d175b222a509
# Parent  a20f86786abbc036c014f334eabdba32b1b7eb0a
Implement x- and y-only panning for FLTK (bug #44408).

* __init_fltk__.cc (handle): get pan mode from figure before calling
translate_view function.

diff --git a/libinterp/dldfcn/__init_fltk__.cc b/libinterp/dldfcn/__init_fltk__.cc
--- a/libinterp/dldfcn/__init_fltk__.cc
+++ b/libinterp/dldfcn/__init_fltk__.cc
@@ -1524,17 +1524,21 @@ private:
                           double x0, y0, x1, y1;
                           Matrix pos = fp.get_boundingbox (true);
                           pixel2pos (ax_obj, pos_x, pos_y, x0, y0);
                           pixel2pos (ax_obj, Fl::event_x (),
                                      Fl::event_y () - menu_dy (),
                                      x1, y1);
 
                           if (pan_enabled ())
-                            ap.translate_view ("both", x0, x1, y0, y1);
+                          {
+                            std::string mode = pan_mode ();
+
+                            ap.translate_view (mode, x0, x1, y0, y1);
+                          }
                           else if (rotate_enabled ())
                             {
                               double daz, del;
                               daz = (Fl::event_x () - pos_x) / pos(2) * 360;
                               del = (Fl::event_y () - menu_dy () - pos_y)
                                     / pos(3) * 360;
                               ap.rotate_view (del, daz);
                             }
